packages/blade/src/utils/styled-props/useStyledProps.native.tsx

```diff
@@ -0,0 +1,12 @@
+import type { CSSObject } from 'styled-components';
+import { getStyledProps } from './getStyledProps';
+import { getBaseBoxStyles } from '~components/Box/BaseBox/getBaseBoxStyles';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const useStyledProps = (props: Record<string, any>): CSSObject => {
+  const styledPropsStyles = getStyledProps(props);
+  const styledPropsCSSObject = getBaseBoxStyles({ ...styledPropsStyles, theme: props.theme });
```

Comment:
Why is this different for web & native?

Reply:
can't use `React.useMemo` on native

Reply:
seriously? 😮

Reply:
oh let me check again. I remember adding it and it breaking. Not sure if broke because useMemo or I messed up somewhere else 😆

Reply:
You can use `useMemo` on native too

Reply:
oh right the problem isn't useMemo.

So in web styled-components, you can use hooks inside styled-components. But in native, you can't use hooks inside styled-components component. 🤔 Let me see if there is some way around.

Reply:
@saurabhdaware was this resolved? I see there's only one `useStyledProps` now